pinot-broker/src/main/java/org/apache/pinot/broker/routing/builder/BasePartitionAwareRoutingTableBuilder.java

```diff
@@ -18,6 +18,7 @@
  */
 package org.apache.pinot.broker.routing.builder;
 
+import it.unimi.dsi.fastutil.ints.IntArrays;
```

Comment:
Can we use ArrayList and use Collections.shuffle() instead of relying on this library?

Reply:
We use this library for high performance code (use primitive type instead of Object). Both ways are okay because it's query level operations, but we can still call this thousands of times per second.